src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVVariantCallerInternal.java

```diff
@@ -0,0 +1,338 @@
+                             .id(produceVariantId(breakpointAllele));
+        vcBuilder = updateAttributes(vcBuilder, breakpointAllele, assembledBreakpointsPerAllele._2, start, end);
+
+        return vcBuilder.make();
+    }
+
+    // -----------------------------------------------------------------------------------------------
+    // Group 1: from contig alignments to breakpoints
+    // -----------------------------------------------------------------------------------------------
+
+    @VisibleForTesting
+    static List<ChimericAlignment> getChimericAlignmentsFromAlignmentRegions(final byte[] sequence, final List<AlignmentRegion> alignmentRegionList, final Integer minAlignLength) {
+        if (alignmentRegionList.isEmpty()) {
+            return new ArrayList<>();
+        }
+        final List<ChimericAlignment> results = new ArrayList<>(alignmentRegionList.size() - 1);
+        final Iterator<AlignmentRegion> iterator = alignmentRegionList.iterator();
+        final List<String> insertionAlignmentRegions = new ArrayList<>();
+        if ( iterator.hasNext() ) {
+            AlignmentRegion current = iterator.next();
+            while (treatAlignmentRegionAsInsertion(current) && iterator.hasNext()) {
+                current = iterator.next();
+            }
+            while ( iterator.hasNext() ) {
+                final AlignmentRegion next = iterator.next();
+                if (alignmentRegionIsTooSmall(current, next, minAlignLength)) {
+                    continue;
+                }
+
+                if (treatNextAlignmentRegionInPairAsInsertion(current, next, minAlignLength)) {
+                    if (iterator.hasNext()) {
+                        insertionAlignmentRegions.add(next.toPackedString());
+                        continue;
+                    } else {
+                        break;
+                    }
+                }
+
+                final AlignmentRegion previous = current;
+                current = next;
+
+                final byte[] sequenceCopy = Arrays.copyOf(sequence, sequence.length);
+
+                String homology = getHomology(current, previous, sequenceCopy);
+                String insertedSequence = getInsertedSequence(current, previous, sequenceCopy);
+
+                final ChimericAlignment chimericAlignment = new ChimericAlignment(previous, current, insertedSequence, homology, insertionAlignmentRegions);
+
+                results.add(chimericAlignment);
+            }
+        }
+        return results;
+    }
+
+    @VisibleForTesting
+    static boolean alignmentRegionIsTooSmall(final AlignmentRegion current, final AlignmentRegion next, final Integer minAlignLength) {
+        return current.referenceInterval.size() - SVVariantCallerUtils.overlapOnContig(current, next) < minAlignLength;
+    }
+
+    // TODO: think about how to test
+    static boolean treatAlignmentRegionAsInsertion(final AlignmentRegion next) {
```

Comment:
done. also updated the name of the method 
`boolean treatNextAlignmentRegionInPairAsInsertion(final AlignmentRegion current, final AlignmentRegion next, final Integer minAlignLength)`